packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ParameterProvider.cs

```diff
@@ -30,6 +30,10 @@ public ParameterProvider(InputModelProperty inputProperty)
             Name = inputProperty.Name.ToVariableName();
             Description = FormattableStringHelpers.FromString(inputProperty.Description);
             Type = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(inputProperty.Type);
+            if (!inputProperty.IsRequired && !Type.IsCollection)
```

Comment:
Why wouldn't this logic be in the type factory?

Reply:
Optionality is a property of a "property", it shouldn't be a property of a "type". You probably missed a truth here: we just leverage `x?` to represent an optional property, it's intrinsically type of `x` which is optional. Therefore, I think `IsRequired` shouldn't be in the type factory. As for `IsCollection`, it's meaningless to put it into typefactory without `IsRequired`.

Reply:
ah I see we are setting nullable due to optional which is indeed a property of a property not of a type.